### PR TITLE
Added tests for chunked encoding

### PR DESCRIPTION
**Pull request by dyrock**
add tests for POST using chunked encoding over HTTP/1.1 and HTTP/2
add test to exercise HTTP/1.1 GET with chunked encoding

### CODE REVIEW

**Comment by bryancall** [comment]
[approve ci]

**Comment by shinrich** [comment]
[approve ci]

File: tests/gold_tests/chunked_encoding/chunked_encoding.test.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+})
+
+ts.Disk.remap_config.AddLine(
+    'map http://www.example.com http://127.0.0.1:{0}'.format(server.Variables.Port)
+)
+ts.Disk.remap_config.AddLine(
+    'map http://www.yetanotherexample.com http://127.0.0.1:{0}'.format(server3.Variables.Port)
+)
+ts.Disk.remap_config.AddLine(
+    'map https://www.anotherexample.com https://127.0.0.1:{0}'.format(server2.Variables.Port, ts.Variables.ssl_port)
```
Comment (shinrich) [inline tests/gold_tests/chunked_encoding/chunked_encoding.test.py]: Do you have an extra argument for the format?

**Comment by shinrich** [comment]
Looks ok to me.  Might be nice to have a big commit at the top of the file, explaining your cases and expectations of each.  Actually Jason probably has some description field on the test cases. 

Actually most of our existing autests could benefit from such comments as well.

**Comment by bryancall** [comment]
[approve ci autest]

**Pull request merged**